OpenRA.Mods.Common/Traits/Render/WithTurretedAttackAnimation.cs

```diff
@@ -0,0 +1,110 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	public class WithTurretedAttackAnimationInfo : ITraitInfo, Requires<WithSpriteTurretInfo>, Requires<ArmamentInfo>, Requires<AttackBaseInfo>
+	{
+		[Desc("Armament name")]
+		public readonly string Armament = "primary";
+
+		[Desc("Turret name")]
+		public readonly string Turret = "primary";
+
+		[Desc("Displayed while attacking.")]
+		[SequenceReference] public readonly string AttackSequence = null;
+
+		[Desc("Displayed while targeting.")]
+		[SequenceReference] public readonly string AimSequence = null;
+
+		[Desc("Shown while reloading.")]
+		[SequenceReference(null, true)] public readonly string ReloadPrefix = null;
+
+		[Desc("Delay in ticks before animation starts, either relative to attack preparation or attack.")]
+		public readonly int Delay = 0;
+
+		[Desc("Should the animation be delayed relative to preparation or actual attack?")]
+		public readonly AttackDelayType DelayRelativeTo = AttackDelayType.Preparation;
+
+		public object Create(ActorInitializer init) { return new WithTurretedAttackAnimation(init, this); }
+	}
+
+	public class WithTurretedAttackAnimation : ITick, INotifyAttack
+	{
+		readonly WithTurretedAttackAnimationInfo info;
+		readonly AttackBase attack;
+		readonly Armament armament;
+		readonly WithSpriteTurret wst;
+
+		int tick;
+
+		public WithTurretedAttackAnimation(ActorInitializer init, WithTurretedAttackAnimationInfo info)
+		{
+			this.info = info;
+			attack = init.Self.Trait<AttackBase>();
+			armament = init.Self.TraitsImplementing<Armament>()
+				.Single(a => a.Info.Name == info.Armament);
```

Comment:
This is problematic - elite weapons are commonplace in TS and RA2. WST is much more likely to stay.